script/modelling/facility_location.jl

```diff
@@ -0,0 +1,235 @@
+#' ---
+#' title: Facility Location
+#' ---
+
+#' **Originally Contributed by**: Mathieu Tanneau and Alexis Montoison
+
+#'Benchmark instances: http://resources.mpi-inf.mpg.de/departments/d1/projects/benchmarks/UflLib/
+
+#' ## Uncapacitated facility location
+#'
+#' Problem description
+#' * $M=\{1, \dots, m\}$ clients, $N=\{ 1, \dots, n\}$ sites where a facility can be built.
+
+n = 2  # number of facility locations
+m = 5  # number of clients
+
+#' * $f_j$: fixed cost of building a facility at site $j$
+#' * $c_{i, j}$: cost for serving customer $i$ from facility $j$
+
+# Draw costs
+Random.seed!(0)
+f = rand(1000:10000, n);
+c = rand(0:1000, m, n);
+
+#' ### MILP formulation
+#'
+#' $$
+#' \begin{array}{cl}
+#' \min_{x, y} \ \ \ &
+#' \sum_{i, j} c_{i, j} x_{i, j} + 
+#' \sum_{j} f_{j} y_{j}\\
+#' s.t. &
+#' \sum_{j} x_{i, j} = 1, \ \ \ \forall i \in M\\
+#' & x_{i, j} \leq y_{j}, \ \ \ \forall i \in M, j \in N\\
+#' & x_{i, j}, y_{j} \in \{0, 1\}, \ \ \ \forall i \in M, j \in N
+#' \end{array}
+#' $$
+
+using JuMP
+using GLPK
+using Cbc
+
+# Create a model
+ufl = Model();
+
+#+
+
+# Create our variables
+@variable(ufl, y[1:m], Bin);
+@variable(ufl, x[1:m, 1:n], Bin);
+
+#+
+
+# Add constraints
+
+# Each client is serve exactly once
+@constraint(
+    ufl,
+    [i in 1:m],
+    sum(x[i, j] for j in 1:n) == 1
+);
+
+# Fixed cost of opening facilities
+@constraint(
+    ufl,
+    [i in 1:m, j in 1:n],
+    x[i, j] <= y[j]
+);
+
+#+
+
+# Set objective
+F = sum([f[j]*y[j] for j in 1:n]);
```

Comment:
Is there a difference between computing the sum like this and using `@expression`?